### PR TITLE
Replace isipv6 method with builtin package

### LINKED ISSUE #1686: Replace an "isIPv6Address()" with python package "ipaddress"
Replace this method `isIPv6Address()` with Python in-build [ipaddress](https://docs.python.org/3/library/ipaddress.html) package introduce with Python 3.3.

https://github.com/bit-team/backintime/blob/d33f35e10b4c3027b3b12881152c157eb1b2cc8c/common/tools.py#L1849

This is the only place this function is used. Remove the code and its unit tests.

### PR DESCRIPTION
**Pull request by jarbhav**
Closes #1686 
- Removed `isIPv6Address()`  function and related tests
- Using `ipaddress` package directly in the `escapeIPv6Address` function

### CODE REVIEW

**Comment by buhtz** [comment]
Dear Vaibhav,
Thank you for your contribution. I appreciate it.

Looks nice. Good work. I added some review comments to the code.

Beside my comments it would be nice if you could improve tests. The behavior of the function `escapeIPv6Address()` has three outcomes: 1) escaped IP 2) not escaped IP 3) raised ValueError. This three outcomes could be covered in three different unit tests. The current test do cover only 1) and 2) in one function but it would improve test quality if each of the three outcomes are tested in separate test methods.

Let me know if I can be of assistance.

Best,
Christian

**Comment by jarbhav** [comment]
Hey Christian, I am unable to see the review comments.
I will get started on the test cases.

**Comment by buhtz** [comment]
Dear Vaibhav,

On 2024-04-15 09:19 Vaibhav Raj ***@***.***> wrote:
> Hey Christian, I am unable to see the review comments.
> I will get started on the test cases.

I wonder how this could be. Do you read via email or do you visit
github.com? When you visit github.com and visit your on PR you will see
something like this (see attached screenshot). I added two circles with
numbers to the picture to point you to the comments.

Can't you see this on the real github.com website?

Kind
Christian

**Comment by jarbhav** [comment]
image

**Comment by buhtz** [comment]
Your screenshot looks like the "Files changed" section. My apologize. I think the term I used (code comments) is not correct. Correct would be review comments. Have a look at this screencast. Does this help you?
Peek 2024-04-15 20-02


But you also should see this review comments in the "Files changed" section when scrolling down.
grafik

**Comment by buhtz** [comment]
Dear Vaibhav,
can I somehow be of assistance?
Kind,
Christian

File: common/tools.py [hunk 1]
```diff
@@ -1821,21 +1821,6 @@ def splitCommands(cmds, head = '', tail = '', maxLength = 0):
...
-        bool:           True if ``address`` is a valid IPv6 address
-    """
-    try:
-        return isinstance(ipaddress.IPv6Address(address), ipaddress.IPv6Address)
-    except:
-        return False
-
 def escapeIPv6Address(address):
     """
     Escape IPv6 Addresses with square brackets ``[]``.
```
Comment (buhtz) [inline common/tools.py]: Please modify the doc string to something like this

```
    """Escape IP addresses with square brackets ``[]`` if they are IPv6.

    If IPv4 nothing is changed.

    Args:
        address (str): IP-Address to escape if needed.

    Returns:
        str: The address, escaped if it is IPv6.

    Raises:
        ValueError: If address is not a valid IPv4 or IPv6.
    """
```

File: common/tools.py [hunk 2]
```diff
@@ -1846,10 +1831,13 @@ def escapeIPv6Address(address):
...
-    if isIPv6Address(address):
-        return '[{}]'.format(address)
-    else:
-        return address
+    try:
+        ip = ipaddress.ip_address(address)
+        if ip.version == 6:
+            return '[{}]'.format(address)
+    except ValueError:
+        pass    # Address is not a valid IP address, so return it unchanged
```
Comment (buhtz) [inline common/tools.py]: I wonder if we might lose some error cases when catching an exception here. I would prefer to see exceptions. Please remove the try-catch block.

```suggestion
    ip = ipaddress.ip_address(address)
    if ip.version == 6:
        return '[{}]'.format(address)
```

**Review by buhtz — CHANGES_REQUESTED** [review]

**Comment by buhtz** [comment]
Dear Vaibhav,
can you please give us some feedback about your plans and your next steps.
Best,
Christian

**Comment by jarbhav** [comment]
Hey Christian,
I apologize for the lack of response; I will try to get back to you with all the changes above in the coming 3 days.
Please let me know if this should be done sooner.
Vaibhav

**Comment by buhtz** [comment]
Dear Vaibhav,
Thank you for the reply.

> I will try to get back to you with all the changes above in the coming 3 days. Please let me know if this should be done sooner.

That is totally fine for me. Take your time. No need to hurry. I just need to know your schedule to "integrate" your PR into the rest of our workflow and tasks. Sometimes we have contributors throwing in a PR but not finalizing it.

Best,
Christian

**Comment by jarbhav** [comment]
Dear Christian,
I have made the suggested changes, also separated the test method into three different methods. Please let me know if any changes are required

Vaibhav

**Comment by buhtz** [comment]
Dear Vaibhav,
please apologize my late reply. Your modifications looking good. I just did some refactoring and thing we are now ready to merge. I will merge in some days after a cool down period. Please see [What happens after you opened a Pull Request (PR)?](https://github.com/bit-team/backintime/blob/dev/CONTRIBUTING.md#what-happens-after-you-opened-a-pull-request-PR) about how we handle this usually.

- I refactored the unit tests a bit.
- I rewinded my ValueError proposal. Other tests failed because they where using host names (e.g. "localhost").

**Review by buhtz — APPROVED** [review]

**Comment by jarbhav** [comment]
Thanks Christian

**Pull request merged**